### PR TITLE
Add the possibility to plot in MJD the light curves

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
This pull request is associated to the Issue #4258 . It adds an argument in the plot function of the FluxPoints, `time_format`. This argument is propagated to the `NDRegionMap` and the `TimeMapAxis` is correctly formatted for mjd.

### CODE REVIEW

File: gammapy/estimators/points/core.py [hunk 1]
```diff
@@ -482,6 +482,8 @@ def plot(self, ax=None, sed_type=None, energy_power=0, **kwargs):
             Sed type
         energy_power : float
             Power of energy to multiply flux axis with
+        time_format : {"iso", "mjd"}
+            Default: "iso"
```
Comment (adonath) [inline gammapy/estimators/points/core.py:486]: Maybe just clarify that this only applies if a time axis is present...

File: gammapy/maps/region/ndmap.py [hunk 2]
```diff
@@ -82,6 +84,8 @@ def plot(self, ax=None, axis_name=None, **kwargs):
                 )
 
         axis = self.geom.axes[axis_name]
+        if isinstance(axis, TimeMapAxis):
+            axis.time_format = time_format
```
Comment (adonath) [inline gammapy/maps/region/ndmap.py:88]: This will change the time format permanently on the object, not just for plotting. I'll make a suggestion in a different comment...

**Review by adonath — COMMENTED** [review]
Thanks @bkhelifi! I think should treat the `TimeMapAxis` as default. So I suggest a behavior along the lines of:

```python

def format_plot_axis(..., time_format=None):
    if time_format is None:
        time_format = self.time_format
```

File: gammapy/maps/region/ndmap.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def __init__(self, geom, data=None, dtype="float32", meta=None, unit=""):
         self.meta = meta
         self._unit = u.Unit(unit)
 
-    def plot(self, ax=None, axis_name=None, **kwargs):
+    def plot(self, ax=None, axis_name=None, time_format="iso", **kwargs):
```
Comment (AtreyeeS) [inline gammapy/maps/region/ndmap.py:55]: Since our Maps are very general, I dont know how nice it is to have explicit arguments for configuring each axis type... Maybe having another `kwargs` would be better?

Comment (adonath) [reply gammapy/maps/region/ndmap.py:55]: So you propose something along the lines of:

```python3
map.plot(axis_name="time", axis_plot_kwargs={"time_format": "mjd"} )
``` 
Where `axis_plot_kwargs` would be forwarded to `axes[axis_name].format_plot_xaxis(**axis_plot_kwargs)`.

I tend to agree, except that right now we do not have arguments for e.g. `MapAxis.format_plot_xaxis()`, so right now there is no other uses case than the time format. If we can imagine to have other use cases then it is probably a good idea.

Comment (AtreyeeS) [reply gammapy/maps/region/ndmap.py:55]: @adonath : Yes, we just discussed this, and at present we dont see other use cases. So maybe this can remain as is now, and in future we can move it to kwargs. This should be backwards compatible

File: gammapy/maps/region/ndmap.py (hunk 2 above)
Comment (adonath) [reply gammapy/maps/region/ndmap.py:88]: I see, but isn't it better then to just modify the solution we currently have? So we could change the `time_format` from as class attribute to an instance attribute. This way use can more easily change the time format between to plots. We could also implement a setter to check for the valid formats...
 
I think the implicit change of `MapAxis.time_format` by calling `.plot()` is not a good idea. Then we should rather use the explicit solution:

```
m.geom.axes["time"].time_format = "mjd"
m.plot(axis_name="time")

# instead of

m.plot(axis_name="time", time_format="mjd")
 
```
If we prefer the latter solution in terms of API we have to choose the solution you propose and add the support to all plot methods...

File: gammapy/maps/region/ndmap.py (hunk 3 above)
Comment (AtreyeeS) [reply gammapy/maps/region/ndmap.py:55]: @adonath : Yes, we just discussed this, and at present we dont see other use cases. So maybe this can remain as is now, and in future we can move it to kwargs. This should be backwards compatible

File: gammapy/maps/axes.py [hunk 4]
```diff
@@ -2380,13 +2382,16 @@ def format_plot_xaxis(self, ax):
...
             ax.xaxis.set_major_formatter(DateFormatter("%Y-%m-%d %H:%M:%S"))
-            plt.setp(
-                ax.xaxis.get_majorticklabels(),
-                rotation=30,
-                ha="right",
-                rotation_mode="anchor",
-            )
-
+        elif self.time_format == "mjd":
+            ax.xaxis.set_major_formatter(ticker.StrMethodFormatter("{x:,.5f}"))
```
Comment (registerrier) [inline gammapy/maps/axes.py:2386]: I find the output not so easy to read. Isn't the standard labeling of matplotlib sufficient for this case? 
Maybe the slanted labeling is not necessary as well.

File: gammapy/maps/tests/test_axes.py [hunk 5]
```diff
@@ -763,6 +763,18 @@ def test_map_axis_format_plot_xaxis():
...
 
 
+def test_time_map_axis_plot_center(time_intervals):
+    axis = TimeMapAxis(
+        time_intervals["t_min"],
+        time_intervals["t_max"],
+        time_intervals["t_ref"],
+        name="time",
+    )
+    axis.time_format = "null"
```
Comment (registerrier) [inline gammapy/maps/tests/test_axes.py:773]: Again it would make more sense if this is the code that raises the exception.

**Review by registerrier — COMMENTED** [review]
Thanks @bkhelifi . 

My main comment here is that it is weird to add multiple exception raise and multiple test for the case where `time_format` is incorrect.
It would seem more rational to create a `setter` property that would fail if the input value is incorrect, i.e when running `axis.time_format = "null"`

File: gammapy/maps/axes.py (hunk 4 above)
Comment (bkhelifi) [reply gammapy/maps/axes.py]: If I remove if, the output seems to me not very nice:

Capture d’écran du 2023-04-17 21-52-51

I prefer this one:

Capture d’écran du 2023-04-17 22-02-18

This is a question of taster. I leave you decide...
In any case, I have simplified the formatting. Thanks

File: gammapy/maps/core.py [hunk 6]
```diff
@@ -51,9 +51,9 @@ def __init__(self, geom, data, meta=None, unit=""):
             self._unit = u.Unit(unit)
 
         if meta is None:
-            self.meta = {}
+            self._meta = {}
```
Comment (registerrier) [inline gammapy/maps/core.py:54]: why this change here?

**Review by registerrier — APPROVED** [review]
Thanks @bkhelifi . Looks good. 
I have only one comment, why do you change the `Map.meta` attribute? I suppose this was not intended to go in this PR.

File: gammapy/maps/core.py (hunk 6 above)
Comment (bkhelifi) [reply gammapy/maps/core.py]: fixed

**Review by registerrier — APPROVED** [review]
Thanks @bkhelifi . No more comments from my side.
